### PR TITLE
LPS-198500 Migrate portal-workflow-task-web - main.js to purely javaScript

### PR DESCRIPTION
**Pull request by Stephy6**
Motivation
=======
The motivation behind this ticket is the migration of js files that use AUI to files that don't use it since is deprecated.
https://liferay.atlassian.net/browse/LPS-198735

Steps to Verify
========
1. Go to Applications Menu -> Applications -> Workflow -> Process Builder -> Configuration -> look for Web Content Article -> Edit it -> change the input to Single Approver -> Save it
2. Go to Product Menu -> Content & Data -> Web Content -> click on (+) -> put a title and some content and click on Submit 
for Workflow
3. Click on the user-avatar link to open the personal menu -> Notifications -> open the Web content review -> click on the dropdown button at the up and right corner  -> Assign to me -> enter some chars and click Done (this will keep you in the same view)
4. In the same view click over the same dropdown button again and check that the Approve and Reject actions work correctly

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: modules/apps/portal-workflow/portal-workflow-task-web/src/main/resources/META-INF/resources/js/Main.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+				title: icon.text,
+			},
+			document.createElement('div')
+		);
+	};
+
+	for (let i = 0; i < workflowTasks.length; i++) {
+		const taskLink = `${portletNamespace}${randomId}${workflowTasks[i]}taskChangeStatusLink`;
+		const element = document.getElementById(taskLink);
+		element?.addEventListener('click', onTaskClickFn);
```
Comment (victorg1991) [inline modules/apps/portal-workflow/portal-workflow-task-web/src/main/resources/META-INF/resources/js/Main.js:34]: We don't need to do a classic loop for this, we can simply use the forEach function in the workflowTasks

Also, the same way that the add the event we need to remove it, using the dispose function available :) example of this here: https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/js/AddSiteNavigationMenuItem.js#L56-L60

**Comment by victorg1991** [comment]
Hey @Stephy6 This looks great :) I have see that you have to create a specific react component for the modal. We have a function available globally for rendering a modal with an input [openSimpleInputModal](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/commands/OpenSimpleInputModal.es.js) We can't use it right now because it render a normal input not a textarea. I was thinking about adding a new argument to that function to allow user to display a modal with a textarea. What do you think? This way we won't need any extra react component :)

**Comment by Stephy6** [comment]
> Hey @Stephy6 This looks great :) I have see that you have to create a specific react component for the modal. We have a function available globally for rendering a modal with an input [openSimpleInputModal](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/commands/OpenSimpleInputModal.es.js) We can't use it right now because it render a normal input not a textarea. I was thinking about adding a new argument to that function to allow user to display a modal with a textarea. What do you think? This way we won't need any extra react component :)

Yeah sure!

**Comment by victorg1991** [comment]
This works like a charm @Stephy6 :clap: 

Would you mind sending this to frontend infra so that they review the changes in simpleInputModal?

Thanks!!

**Comment by Stephy6** [comment]
> This works like a charm @Stephy6 👏
> 
> Would you mind sending this to frontend infra so that they review the changes in simpleInputModal?
> 
> Thanks!!

On my way!

**Comment by victorg1991** [comment]
Closing this, since you already sent it to FI https://github.com/liferay-frontend/liferay-portal/pull/3764

**Pull request closed without merging**